modules/tensor_mechanics/src/kernels/lagrangian/LagrangianStressDivergenceBase.C

```diff
@@ -0,0 +1,146 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LagrangianStressDivergenceBase.h"
+
+InputParameters
+LagrangianStressDivergenceBase::validParams()
+{
+  InputParameters params = Kernel::validParams();
+
+  params.addRequiredParam<unsigned int>("component", "Which direction this kernel acts in");
+  params.addRequiredCoupledVar("displacements", "The displacement components");
+
+  params.addParam<bool>("large_kinematics", false, "Use large displacement kinematics");
```

Comment:
Would it make sense to use an enumeration like `kinematics=LARGE|SMALL` to be consistent how we do `strain=SMALL|FINITE` in the TM master action? I'm not sure whether this is worth the trouble, but it might be.

Reply:
I prefer a `bool`, as logically there can be only two options (right?).  On the other hand, if you think of a case where we might want to expand to 3 or more options in the future then I'm fine changing it.

Reply:
I guess the advantage of the enum is it makes it more clear what the two options are, although I guess it's pretty apparent that 'small' would be what you get if large=false. I'm fine keeping it how you have it.